src/EmployeeBook.java

```diff
@@ -0,0 +1,262 @@
+            }
+            System.out.println(employee[i]);
+        }
+    }
+
+    public int salaryPerMonth(){
+        int total = 0;
+        for (int i = 0; i < employee.length; i++) {
+            if (employee[i] == null){
+                continue;
+            }
+            total += employee[i].getSalary();
+        }
+        return total;
+    }
+    public Employee minimumSalaryCalculation(){
+        int minSalary = Integer.MAX_VALUE;
+        int identifier = 0;
+        for (int i = 0; i < employee.length; i++) {
+
+            if (employee[i] == null){
+                continue;
+            }
+
+            if (minSalary > employee[i].getSalary()){
+                minSalary = employee[i].getSalary();
+                identifier = i;
+            }
+
+
+        }
+        return employee[identifier];
+    }
+    public Employee maximumSalaryCalculation(){
+        int maxSalary = 0;
+        int identifier = 0;
+        for (int i = 0; i < employee.length; i++) {
+
+            if (employee[i] == null){
+                continue;
+            }
+
+            if (maxSalary < employee[i].getSalary()){
+                maxSalary = employee[i].getSalary();
+                identifier = i;
+            }
+
+        }
+        return employee[identifier];
+    }
+    public int calculationOfAverageSalary(){
+        int monthlyEmployeeSalary = 0;
+        int total = 0;
+        for (int i = 0; i < employee.length; i++) {
+            if (employee[i] == null){
+                continue;
+            }
+            total++;
+            monthlyEmployeeSalary += employee[i].getSalary();
+
+        }
+        if(monthlyEmployeeSalary == 0 || total == 0){
+            return 0;
+        }else{
+            return monthlyEmployeeSalary / total;
+        }
+
+    }
+    public void printFullnamesOfEmployees(){
+        System.out.println("ФИО всех сотрудников: ");
+        for (int i = 0; i < employee.length; i++) {
+            if (employee[i] == null){
+                continue;
+            }
+            System.out.println(employee[i].getFamilName() + " " + employee[i].getName() + " " + employee[i].getSurname());
+
+        }
+
+    }
+    ///////////////////// Повышенная сложность
+    public void indexingSalary(int percent){
+        for (int i = 0; i < employee.length; i++) {
+            if (employee[i] == null){
+                continue;
+            }
+            int amountIndexingField = employee[i].getSalary() * percent / 100 + employee[i].getSalary();
+            employee[i].setSalary(amountIndexingField);
+        }
+    }
+    public Employee minSalaryForTheDepartment(int department){
+        int minSalary = Integer.MAX_VALUE;
+        int minIdentifier = 0;
+        for (int i = 0; i < employee.length; i++) {
+            if (employee[i] == null) {
+                continue;
+            }
+            if (department == employee[i].getDepartment()) {
+                if (employee[i].getSalary() < minSalary) {
```

Comment:
```suggestion
            if (department == employee[i].getDepartment() && employee[i].getSalary() < minSalary) {
```
лучше объединить в один if, убрав вложенность